src/loops.rs

```diff
@@ -338,20 +339,28 @@ fn check_for_loop_range(cx: &LateContext, pat: &Pat, arg: &Expr, body: &Expr, ex
     if let ExprRange(Some(ref l), ref r) = arg.node {
         // the var must be a single name
         if let PatIdent(_, ref ident, _) = pat.node {
+
             let mut visitor = VarVisitor {
                 cx: cx,
                 var: ident.node.name,
-                indexed: HashSet::new(),
+                indexed: HashMap::new(),
                 nonindex: false,
             };
             walk_expr(&mut visitor, body);
+
             // linting condition: we only indexed one variable
             if visitor.indexed.len() == 1 {
-                let indexed = visitor.indexed
+                let (indexed, indexed_extent) = visitor.indexed
                                      .into_iter()
                                      .next()
                                      .expect("Len was nonzero, but no contents found");
```

Comment:
we could use `.unwrap_or(|| unreachable!())` here, it would document the unreachability equally well and may be better optimized by the compiler.